### PR TITLE
Application Review step fully functional

### LINKED ISSUE #2891: Task - Application - Make Review page fully functional
# Description
Ensure the review page loads the application all associated data it needs. Ensure there's a way of controlling whether edit links appear (applicaiton vs manager portal) and that they link to the correct place. Ensure the SAVE AND EXIT (which can just exit in this) and CONTINUE links work as well.

# Images
https://xd.adobe.com/view/a678f763-f7b7-43ca-740a-89359cbe42f1-d049/screen/77720656-484b-444f-a990-c6f045d88e51/Step-2B-Requirements-18

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #2891.

### Notes:
- Like other Application steps, this now triggers fetching of all Application data (if its not yet in Redux store).
- The Save and ____ buttons now save the 'share with manager' field of the application.
- This PR also adds the data fetching to the Welcome step of the Application.

### CODE REVIEW

**Review by Khristophor — CHANGES_REQUESTED** [review]
Looks good, just a couple failing unit tests to fix up

File: resources/assets/js/hooks/applicationHooks.tsx [hunk 1]
```diff
@@ -254,8 +261,10 @@ export function useFetchApplication(
   const applicationIsUpdating = useSelector((state: RootState) =>
     getApplicationIsUpdating(state, { applicationId }),
   );
+  const [applicationFetched, setApplicationFetched] = useState(false);
   useEffect(() => {
-    if (application === null && !applicationIsUpdating) {
+    if (application === null && !applicationIsUpdating && !applicationFetched) {
+      setApplicationFetched(true);
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/assets/js/hooks/applicationHooks.tsx:267]: Yes, this prevents more dispatches after a failure. Can you think of any obvious conditions where we'd want to retry?

File: resources/assets/js/components/Application/Review/ReviewPage.tsx [hunk 2]
```diff
@@ -28,53 +36,113 @@ interface ReviewPageProps {
...
+            application,
+            intl,
+            "review",
+          )}
+        />
+      )}
+      {allDataLoaded &&
+      application !== null &&
+      job !== null &&
+      user !== null ? (
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/assets/js/components/Application/Review/ReviewPage.tsx:121]: It does, but typescript can't seem to infer that. Without the additional conditions, TypeScript throws compile errors. :(

[the remark this answers is not in the document]
Comment (Khristophor) [reply resources/assets/js/components/Application/Review/ReviewPage.tsx:121]: 😭

File: resources/assets/js/hooks/applicationHooks.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Khristophor) [reply resources/assets/js/hooks/applicationHooks.tsx:267]: I was just thinking it feels weird that if it attempts one fetch, and it doesn't resolve, there's no real way for a user to reattempt aside from a full refresh. Maybe it's just a messaging thing we could improve on (Hey something went wrong try refreshing the page or coming back later).

Or if it's a timeout and not a 500/403/422, we could reattempt after a specific delay.

[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/assets/js/hooks/applicationHooks.tsx:267]: You're right. I'm just not sure how to accomplish that right now without a substantial change to how we manage state and these requests.

**Review by Khristophor — APPROVED** [review]

**Pull request merged**